static/js/apps/nl_interface/query_context.tsx

```diff
@@ -59,41 +69,96 @@ export function QueryContext(): JSX.Element {
   // prompt.
   const searchDelayTimer = useRef(null);
 
-  // Updates the query search input box value.
-  function updateSearchInput(input: string): void {
-    (document.getElementById("query-search-input") as HTMLInputElement).value =
-      input;
-  }
-
-  // Searches for the query in the query search box.
-  function executeSearch(): void {
-    (
-      document.getElementById("rich-search-button") as HTMLButtonElement
-    ).click();
-  }
+  const inputNextPrompt = useCallback(
+    (delayStart: boolean): void => {
+      const prompt = urlPrompts[autoPlayCurrentQueryIndex];
+      //const nextPromptDelay = 0;
+      const promptSearchDelay = autoPlayDisableTypingAnimation
+        ? 0
+        : PROMPT_SEARCH_DELAY;
+      const nextPromptDelay =
+        delayStart && !autoPlayDisableTypingAnimation
+          ? autoPlayAutoShowQueryDelay
+          : 0;
+      nextPromptDelayTimer.current = setTimeout(() => {
+        // If delay is disabled, input the whole prompt at once
+        let inputLength = autoPlayDisableTypingAnimation ? prompt.length : 1;
+        inputIntervalTimer.current = setInterval(() => {
+          if (inputLength <= prompt.length) {
+            setAutoRunQuery(prompt.substring(0, inputLength));
+          }
+          if (inputLength === prompt.length) {
+            clearInterval(inputIntervalTimer.current);
+            // If on autoPlayAutoRunQuery, search for the current input after
+            // PROMPT_SEARCH_DELAY ms.
+            if (autoPlayAutoRunQuery) {
+              searchDelayTimer.current = setTimeout(() => {
+                setAutoRunQuery("");
+                search({
+                  config,
+                  nlQueryContext: currentNlQueryContext,
+                  nlQueryHistory,
+                  query: prompt,
+                });
+              }, promptSearchDelay);
+            }
+            return;
+          }
+          inputLength++;
+        }, CHARACTER_INPUT_INTERVAL);
+      }, nextPromptDelay);
+    },
+    [
+      autoPlayCurrentQueryIndex,
+      autoPlayDisableTypingAnimation,
+      autoPlayAutoShowQueryDelay,
+      autoPlayAutoRunQuery,
+      currentNlQueryContext,
+      nlQueryHistory,
+      urlPrompts,
+      search,
+      setAutoRunQuery,
+    ]
+  );
+  function inputNextPrompt2(delayStart: boolean): void {
```

Comment:
maybe func-level comment on what this does?

Reply:
Whoops- meant to drop this function all together. Dropped!